pkg/awsutils/awsutils.go

```diff
@@ -404,13 +404,49 @@ func (cache *EC2InstanceMetadataCache) refreshSGIDs(mac string) error {
 	addedSGs := newSGs.Difference(&cache.securityGroups)
 	deletedSGs := cache.securityGroups.Difference(&newSGs)
 
+	addedSGsCount := 0
+	deletedSGsCount := 0
+
 	for _, sg := range addedSGs.SortedList() {
 		log.Infof("Found %s, added to ipamd cache", sg)
+		addedSGsCount++
 	}
 	for _, sg := range deletedSGs.SortedList() {
 		log.Infof("Removed %s from ipamd cache", sg)
+		deletedSGsCount++
 	}
+
+	log.Infof("AddedSGsCount = %d and DeletedSgsCount = %d", addedSGsCount, deletedSGsCount)
+
 	cache.securityGroups.Set(sgIDs)
+
+	if addedSGsCount != 0 || deletedSGsCount != 0 {
+		var sgIDsPtrs []*string
+		sgIDsPtrs = aws.StringSlice(sgIDs)
+
+		// Fetch all local ENI info from metadata
+		allENIs, err := cache.GetAttachedENIs()
```

Comment:
This will update all ENIs. We need filter out for unmanaged ENI and just update ENIs which are created by CNI here.